include/deal.II/numerics/rtree.h

```diff
@@ -0,0 +1,146 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2017 - 2018 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_numerics_rtree_h
+#define dealii_numerics_rtree_h
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/base/point.h>
+
+#include <deal.II/boost_adaptors/bounding_box.h>
+#include <deal.II/boost_adaptors/point.h>
+#include <deal.II/boost_adaptors/segment.h>
+
+#include <boost/geometry.hpp>
+
+#include <memory>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+/**
+ * A wrapper for the boost::geometry library, used to compute an RTree from a
+ * collection of Point<dim> based spatial objects, BoundingBox<dim>, or
+ * Segment<dim> base spatial objects.
+ *
+ * An RTree may store any type of @p LeafType as long as it is possible to extract
+ * an @p Indexable that the RTree can handle and compare Values. The @p Indexable
```

Comment:
`The @p Indexable` -> `An @p Indexable`

Reply:
I've not found it particularly easy to follow the use of the term `Indexable` throughout this documentation. Might it make more sense to call these `IndexableType`s?

Reply:
I'm not sure. Boost uses Indexable in all its documentation. Changing name here would only make it more confusing... I tried to extend a little bit the explanation at the beginning. Maybe this will make you happier. :)

Reply:
> Boost uses Indexable in all its documentation. Changing name here would only make it more confusing...

Fair enough. Consistency between our documentation and there's would be a good thing.

> I tried to extend a little bit the explanation at the beginning. Maybe this will make you happier. :)

Yes, the additional documentation is really helpful. Thanks!

Reply:
Still, it remains that we need to adjust one of the articles used in this sentence. I don't think that its correct to use both a definite and an indefinite article when referencing the same thing. So this should either read

> The `@p Indexable` is the type adapted to the Point, BoundingBox or Segment concept

or 

> An `@p Indexable` is a type adapted to the Point, BoundingBox or Segment concept

I favour the second option, since it refers to the abstract concept in the same way that you have done in the preceding sentence.